Add and remove feature branch

ClintonYounge opened the pull request:
**Added Features**
- Feature to add and display tasks.
- Feature to remove tasks.
- Local Storage.
- Feature to edit already added tasks.

ClintonYounge committed:
Update html css and js

Update to add-
-Add to list feature
-Remove from list feature
-CSS for the UI

ClintonYounge committed:
Update js and css

-Update js and css to improce styling.

ClintonYounge committed:
Update font style

ClintonYounge committed:
Update style.css

BenMukebo commented on src/module/tasklist.js:
```diff
@@ -0,0 +1,71 @@
+}
+
+// Add a new task to the list
+export function addTask(description) {
+  tasks.push({
+    description,
+    completed: false,
+    index: tasks.length,
+  });
```
- [ ] Great job so far 👏🏻 However the property index should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). Actually, your first book has an index equal to 0; instead of 1. Kindly address this 💯 👍🏻 



- Please visit this [repo ](https://github.com/BahirHakimy/ToDo-List/pull/2)if you have difficulty understanding the requested change 👍🏻

ClintonYounge replied:
Makes sense.

BenMukebo commented on src/module/tasklist.js:
```diff
@@ -0,0 +1,71 @@
+      saveTasks();
+    });
+
+    deleteButton.addEventListener('click', () => {
+      tasks.splice(index, 1);
+      saveTasks();
+      renderTasks();
+    });
```
- [ ] here is the result of your local storage after deleting book3 and book5:

```js
0 {description: "book1", index: 0, completed: true}
1 {description: "boo2", index: 1, completed: true}
2 {description: "Book4", index: 3, completed: false}
3 {description: "Book6", index: 5, completed: false}
```

- [ ]  Make sure that deleting a task update all remaining items' indexes, so they represent the current list order and are unique. 👍🏻

ClintonYounge replied:
Makes sense.

BenMukebo reviewed (changes requested):
### Hi @ClintonYounge, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- Function for adding a new book is implemented. ✔️
- Function for remove a book from the array is implemented. ✔️
- Function for editing a book from the array is implemented. ✔️
- GitHub flow is followed. :heavy_check_mark:
- All changes to the To-Do List  are saved in local storage. :heavy_check_mark:


 ## Required Changes ♻️
_Check the comments under the review._

You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ClintonYounge committed:
Update js

- tasks property index set to the value of the new array length

- deleting a task update all remaining items' indexes

ClintonYounge committed:
Update main.js

OLIPLICHE reviewed (approved):
Hi @ClintonYounge, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Implemented Create, Update, Delete👏 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
